modin/pandas/base.py

```diff
@@ -983,8 +983,19 @@ def astype(self, dtype, copy=True, errors="raise"):  # noqa: PR01, RT01, D200
             # Assume that the dtype is a scalar.
             col_dtypes = {column: dtype for column in self._query_compiler.columns}
 
-        new_query_compiler = self._query_compiler.astype(col_dtypes, errors=errors)
-        return self._create_or_update_from_compiler(new_query_compiler, not copy)
+        if not copy:
+            # If the new types match the old ones, then copying can be avoided
+            frame_dtypes = self._query_compiler._modin_frame._dtypes
+            if frame_dtypes is not None:
```

Comment:
just a notice: we may want eventually to propagate `has_*_materialized` to the qc level
```suggestion
            if self._query_compiler.has_dtypes_materialized:
```